pom.xml

```diff
@@ -284,6 +290,11 @@
       <version>1.9.5</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-compat</artifactId>
+      <version>3.0</version>
+    </dependency>
```

Comment:
This should not been used cause it contains the compatibility layer to Maven 2...so the plugin should work without this one...

Reply:
I found that quite difficult.  Can you help?

Reply:
@epabst I assume @khmarbaise means that maven-compat dependency can be dropped; it's a library which supplies Maven 2-specific classes, which I guess this plugin uses, making it able to run under Maven 3 as well. If the plugin is supposed to drop support for Maven 2 that dependency can be safely removed.

Reply:
Without that dependency,  it won't compile.   I don't know how to make it
compile without it although I tried.

Reply:
I've tried removing the dependency myself and it seems the biggest offender (among others) is WagonManager. According to https://maven.apache.org/ref/3.5.0/apidocs/org/apache/maven/artifact/manager/WagonManager.html, the whole class has been marked for deprecation; you'll therefore need to find a replacement class which fills the same purpose. A starting point is https://maven.apache.org/ref/3.5.0/apidocs/org/apache/maven/repository/legacy/WagonManager.html, it seems to be the 3.0 equivalent.

Reply:
Another alternative could be to refactor the code to use the [Maven Wagon](http://maven.apache.org/wagon/index.html) library with the HTTP provider, but it entails a lot of refactoring. I suggest you wait for input from @khmarbaise.

Reply:
If I correctly see the problem you need maven-artifact-transfer which handled the resolving of artifacts See https://maven.apache.org/shared/maven-artifact-transfer/  but I think one functionality is missing in maven-artifact-transfer to get the list of existing versions of a dependency..(Which I'm currently working on..)...

Reply:
@khmarbaise

> but I think one functionality is missing in maven-artifact-transfer to get the list of existing versions of a dependency..(Which I'm currently working on..)...

Is this done?